components/FilterList/index.js

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from "react";
+import ProjectsList from "../ProjectsList";
+import styled from "styled-components";
+
+export default function FilterList({ projects }) {
+  const complexities = [
+    { name: "All", color: "#070ff7" },
+    { name: "Beginner", color: "#3ecd5e" },
+    { name: "Intermediate", color: "#f9b234" },
+    { name: "Advanced", color: "#e44002" },
+  ];
+
+  const [activeFilter, setActiveFilter] = useState("All");
+
+  const filteredProjects =
+    activeFilter === "All"
+      ? projects
+      : projects.filter((project) => project.complexity === activeFilter);
+
+  function filterByComplexity(complexity) {
+    setActiveFilter(complexity);
+  }
+  return (
+    <>
+      <StyledButtonWrapper>
+        {complexities.map((complexity) => (
+          <StyledButton
+            key={complexity.name}
+            onClick={() => filterByComplexity(complexity.name)}
+            color={complexity.color}
+            isActive={activeFilter === complexity.name}
```

Comment:
Since this prop is only for styling purposes you should prefix it with an '$'. ('$isActive'). (See the 'styled-component' handout for more information)